app/lib/three_scale/api/collection.rb

```diff
@@ -1,42 +1,67 @@
-class ThreeScale::Api::Collection
-
-  include PaginationHelper::ClassMethods
+# frozen_string_literal: true
 
+class ThreeScale::Api::Collection
   def initialize(collection, options = {})
     @collection = collection
     @options = options
   end
 
-  def to_xml(options = {})
-    builder = options[:builder] || ThreeScale::XML::Builder.new
+  def to_xml(xml_options = {})
+    builder = xml_options[:builder] || ThreeScale::XML::Builder.new
+    options.merge!(xml_options).merge!(builder: builder, skip_instruct: true)
 
-    root = options.delete(:root) || @options.delete(:root)
-    # merge options passed when creating collection with the ones passed now together with builder
-    options = options.merge(@options).merge(builder: builder, skip_instruct: true)
-
-    builder.tag!(root, pagination) do |xml|
-      each do |item|
-        item.to_xml(options)
-      end
+    builder.tag!(options.delete(:root), pagination_metadata) do |xml|
+      represented.each { |item| item.to_xml(options) }
     end
 
     builder.to_xml
   end
 
+  def to_json(json_options = {})
+    MultiJson.dump collection.to_hash(json_options).merge(pagination_metadata(wrapper: :metadata))
+  end
+
+  # rubocop:disable Lint/MissingSuper
   def method_missing(method_name, *arguments, &block)
-    collection = @collection.try(:represented) || @collection
-    collection.send(method_name, *arguments, &block)
+    represented.send(method_name, *arguments, &block)
   end
+  # rubocop:enable Lint/MissingSuper
 
   # http://robots.thoughtbot.com/post/28335346416/always-define-respond-to-missing-when-overriding
-
+  # :reek:BooleanParameter ThreeScale::Api::Collection#respond_to_missing? has boolean parameter 'include_private'
+  # :reek:ManualDispatch ThreeScale::Api::Collection#respond_to_missing? manually dispatches method call
+  # rubocop:disable Lint/MissingSuper
   def respond_to_missing?(method_name, include_private = false)
-    @collection.respond_to?(method_name, include_private)
+    collection.respond_to?(method_name, include_private)
   end
+  # rubocop:enable Lint/MissingSuper
 
   private
 
-  def pagination
-    pagination_attrs(@collection)
+  PAGINATION_METADATA = %i[per_page total_entries total_pages current_page].freeze
+  private_constant :PAGINATION_METADATA
+
+  attr_reader :collection, :options
+
+  def represented
+    @represented ||= collection.try(:represented) || collection
+  end
+
+  def pagination_metadata(wrapper: nil)
+    return {} unless paginated?
+
+    wrapper ? {wrapper => pagination_attrs} : pagination_attrs
+  end
+
+  def pagination_attrs
+    PAGINATION_METADATA.each_with_object({}) do |meta_field, pagination_hash|
+      pagination_hash[meta_field] = represented.public_send(meta_field)
+    end
+  end
+
+  def paginated?
+    represented.per_page < represented.total_entries
+  rescue NoMethodError
+    false
```

Comment:
Is this possible?
```suggestion
    return false unless represented.respond_to?(:per_page, :total_entries)

    represented.per_page < represented.total_entries
```

Reply:
something similar is possible, but imo it is not worth it, i prefer to just late it raise instead of checking if it has it or not

Reply:
by similar i mean: `return false unless represented.respond_to?(:per_page) && represented.respond_to?(:total_entries)`

Reply:
I see, better catch the error then 👍